src/webview/leetCodeSolutionProvider.ts

```diff
@@ -1,69 +1,42 @@
 // Copyright (c) jdneo. All rights reserved.
 // Licensed under the MIT license.
 
-import { Disposable, ExtensionContext, ViewColumn, WebviewPanel, window } from "vscode";
+import { ViewColumn } from "vscode";
 import { IProblem } from "../shared";
+import { ILeetCodeWebviewOption, LeetCodeWebview } from "./LeetCodeWebview";
 import { markdownEngine } from "./markdownEngine";
 
-class LeetCodeSolutionProvider implements Disposable {
+class LeetCodeSolutionProvider extends LeetCodeWebview {
 
-    private context: ExtensionContext;
-    private panel: WebviewPanel | undefined;
-
-    public initialize(context: ExtensionContext): void {
-        this.context = context;
-    }
+    private solution: Solution;
 
     public async show(solutionString: string, problem: IProblem): Promise<void> {
-        if (!this.panel) {
-            this.panel = window.createWebviewPanel("leetCode.solution", "Top Voted Solution", ViewColumn.Active, {
-                retainContextWhenHidden: true,
-                enableFindWidget: true,
-                localResourceRoots: markdownEngine.localResourceRoots,
-            });
-
-            this.panel.onDidDispose(() => {
-                this.panel = undefined;
-            }, null, this.context.subscriptions);
+        this.solution = this.parseSolution(solutionString);
```

Comment:
Same problem. If in the future, `getWebviewOption()` is called before `this.solution`, then we could get undefined error